datahub-web-react/src/app/search/context/SearchContextProvider.tsx

```diff
@@ -0,0 +1,31 @@
+import * as QueryString from 'query-string';
+import { useHistory, useLocation } from 'react-router';
+import React, { useEffect, useMemo, useState } from 'react';
+import { SearchContext } from './SearchContext';
+import { updateUrlParam } from '../utils/utils';
+import { DEFAULT_SORT_OPTION, SORT_OPTIONS } from './constants';
+
+export default function SearchContextProvider({ children }: { children: React.ReactNode }) {
+    const [selectedSortOption, setSelectedSortOption] = useState(DEFAULT_SORT_OPTION);
```

Comment:
Not sure if I'm missing something, but why do we need another state value here? Couldn't we just map what we need from  `params` into another var and pass that into the context? so updateSelectedSortOption would mutate the url and then params would swap the value for a fresh context object.

Reply:
yeah tbh this is because I built it without storing in the URLfirst to start and then added storing in the URL later - but now with fresh eyes I totally agree. I like having this context as it's easier to grab values and set values and we can extend it later for other global search state stuff.

so I think i'll just remove this local state and pass in `selectedSortOption` as the value from URL params and the setter function to update url params.